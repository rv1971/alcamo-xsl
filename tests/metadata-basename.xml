<?xml version="1.0" encoding="UTF-8"?>

<?xml-stylesheet href="#xslt" type="text/xml"?>

<a:doc xmlns:a="tag:rv1971@web.de,2021:alcamo-xsl#">
  <xsl:stylesheet
      xmlns:xsl="http://www.w3.org/1999/XSL/Transform"
      xml:id="xslt"
      version="1.0">
    <xsl:import href="../xsl/metadata.xsl"/>

    <xsl:template match="/">
      <a:result>
        <xsl:apply-templates select="/a:doc/a:data"/>
     </a:result>
    </xsl:template>

    <xsl:template match="a:item">
      <a:item>
        <xsl:value-of select="."/>
        <xsl:value-of select="' '"/>
        <xsl:apply-templates select="." mode="a:basename"/>
        <xsl:value-of select="' '"/>

        <xsl:apply-templates select="." mode="a:basename">
          <xsl:with-param name="sep" select="'f'"/>
        </xsl:apply-templates>

        <xsl:value-of select="' '"/>
        <xsl:apply-templates select="." mode="a:dirname"/>
        <xsl:value-of select="' '"/>

        <xsl:apply-templates select="." mode="a:dirname">
          <xsl:with-param name="sep" select="'f'"/>
        </xsl:apply-templates>
      </a:item>
    </xsl:template>
  </xsl:stylesheet>

  <a:data>
    <a:item>foo</a:item>
    <a:item>file:///foo/bar/baz.xml</a:item>
    <a:item>qux/fox/</a:item>
  </a:data>
</a:doc>
