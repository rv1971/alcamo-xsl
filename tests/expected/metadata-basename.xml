<?xml version="1.0"?>
<a:result xmlns:a="tag:rv1971@web.de,2021:alcamo-xsl#">
  <a:item>foo foo oo  f</a:item>
  <a:item>file:///foo/bar/baz.xml baz.xml oo/bar/baz.xml file:///foo/bar/ file:///f</a:item>
  <a:item>qux/fox/  ox/ qux/fox/ qux/f</a:item>
</a:result>
